clusterDeployer.py

```diff
@@ -41,13 +43,20 @@ def match_to_proper_version_format(version_cluster_config: str) -> str:
 
 
 class ClusterDeployer(BaseDeployer):
-    def __init__(self, cc: ClustersConfig, ai: AssistedClientAutomation, steps: list[str], secrets_path: str):
+    def __init__(self, cc: ClustersConfig, ai: AssistedClientAutomation, steps: list[str], secrets_path: str, cdaConfig: CdaConfig, should_resume: bool = False):
         super().__init__(cc, steps)
         self.bf_connections: dict[str, host.Host] = {}
         self._client: Optional[K8sClient] = None
         self._ai = ai
         self._secrets_path = secrets_path
 
+        self.state = StateFile(cdaConfig)
+        reset_state = self.state.state_dict.get("cluster-name") != cc.name or not should_resume
```

Comment:
` self.state.state_dict.get` ->  https://en.wikipedia.org/wiki/Law_of_Demeter
You're leaking implementation details from the StateFile class. Can you instead implement `StateFile.some_func()` where some_func has a clear name telling me what we're trying to achieve, not how we're trying to achieve it.
Also, not clear to me how you support multiple clusters being deployed on the same node. 

Note that currently we currently deploy 2 clusters in parallel. https://github.com/openshift/dpu-operator/blob/main/hack/both.sh#L11-L13

We need a way to resume either